crates/next-core/src/next_app/metadata/route.rs

```diff
@@ -145,8 +145,9 @@ async fn static_route_source(mode: NextMode, path: FileSystemPath) -> Result<Vc<
 
     let original_file_content_b64 = get_base64_file_content(path.clone()).await?;
 
-    let is_twitter = stem == "twitter-image";
-    let is_open_graph = stem == "opengraph-image";
+    // Use 'contains' since they can have suffixes.
+    let is_twitter = stem.contains("twitter-image");
```

Comment:
`contains` seems not safe, `f2twitter-image234` can pass. it should be `twitter-image\d?`. can also change this in a separate PR if this requires more changes